lm_eval/evaluator.py

```diff
@@ -297,13 +293,9 @@ def evaluate(
         versions[task_name] = task.VERSION
         configs[task_name] = dict(task.dump_config())
 
-        if "num_fewshot" in configs[task_name]:
-            if configs[task_name]["metadata"]:
-                n_shot = configs[task_name]["metadata"].get("num_fewshot", None)
-            if not n_shot:
-                n_shot = configs[task_name]["num_fewshot"]
-        else:
-            n_shot = 0 # TODO: is this always right?
+        if (n_shot := configs[task_name].get("num_fewshot")) is None:
```

Comment:
This previously errored out with `squadv2`. Also I changed the priority of n_shot from "metadata" to "num_fewshot". I think that's what we want?

Reply:
The original idea is meant for tasks which have `num_fewshot` hardcoded to 0 (which will disallow changing `num_fewshot` through the `--num_fewshot` arg) but is actually has some fewshot samples that were appended through the `description`, for example BBH which is 3-shot CoT. So we would want to check both if `num_fewshot` is 0 and `num_fewshot` is in `metadata`.

Correct me if I'm wrong @haileyschoelkopf

Reply:
Ok I changed it to check if `num_fewshot == 0` first and then check for `metadata`